RIOT-OS/RIOT
# gnrc_lorawan: encode LoRaWAN port in netif header

### Contribution description
This PR encodes the LoRaWAN port in a GNRC netif header.
The benefit of doing this is that we can reuse `txtsnd` and similar commands. From now on it's possible to use GNRC LoRaWAN (almost) out of the box in any GNRC application (e.g `examples/default`).

This PR also deprecates `NETOPT_LORAWAN_TX_PORT` and introduces a `CONFIG_GNRC_NETIF_LORAWAN_NETIF_HDR` to enable this functionality (and don't abruptly change the API of users).




### Testing procedure

The following things should be tested:
- Check that `examples/gnrc_lorawan` works out of the box. The `send` command was replaced by `txtsnd`
- Try sending frames in port 0. Frame shouldn't be sent.
- Try GNRC LoRaWAN with `examples/default`:
```
CFLAGS=-DCONFIG_GNRC_NETIF_LORAWAN_NETIF_HDR USEMODULE=gnrc_lorawan BOARD=b-l072z-lrwan1 make flash term
```
Check that's possible to communicate.




### Issues/PRs references
None so far
